### PR TITLE
upgrades to admin: search_fields, list_filters and raw_id_field

### PR DESCRIPTION
**Pull request by PetrDlouhy**
Fixes:

* long loading times of admin edit pages when many User/Token models

## Description of the Change

I defined `search_fields`, `list_filters` and `raw_id_fields` for easier management of the tokens in admin.

Not sure what to test in this.
Also I think, this doesn't need any documentation, it is pretty self-explanatory to users.

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [ ] unit-test added
- [ ] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by PetrDlouhy** [comment]
This is my second take on this, I fixed the User without e-mail problem, that was in #723

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
thank you

**Comment by n2ygk** [comment]
In hindsight, it would probably be a good thing for the Admin UI to be documented somewhere....

**Comment by PetrDlouhy** [comment]
@n2ygk Which part of the Admin UI do you think should be documented. Do you mean the UI as general? Or just the search by e-mail functionality (which don't work if there is no e-mail)?

**Comment by n2ygk** [comment]
@PetrDlouhy not here, but as an update to the Tutorial it would probably be helpful to show users around the admin UI. I'll make an issue.

File: oauth2_provider/admin.py [hunk 1]
```diff
@@ -38,11 +44,15 @@ class GrantAdmin(admin.ModelAdmin):
 class IDTokenAdmin(admin.ModelAdmin):
     list_display = ("jti", "user", "application", "expires")
     raw_id_fields = ("user",)
+    search_fields = ("token",) + (("user__email",) if has_email else ())
```
Comment (danlamanna) [inline oauth2_provider/admin.py:47]: Is this valid? Searching from the admin now gives me `django.core.exceptions.FieldError: Cannot resolve keyword 'token' into field. Choices are: access_token, application, application_id, created, expires, id, jti, scope, updated, user, user_id`

**Comment by PetrDlouhy** [comment]
@danlamanna You are right. Fixed in PR #1085

**Comment by charettes** [comment]
FWIW it might have been better to search by `USERNAME_FIELD` of the user model instead as nothing guarantees that the email field will be indexed `USERNAME_FIELD` must be to the unique constraint defined on it.

**Comment by n2ygk** [comment]
> FWIW it might have been better to search by `USERNAME_FIELD` of the user model instead as nothing guarantees that the email field will be indexed while `USERNAME_FIELD` must be to support the unique constraint defined on it.

Yes, that makes sense @charettes. A PR to change this would be appreciated.

**Comment by charettes** [comment]
@n2ygk done in https://github.com/jazzband/django-oauth-toolkit/pull/1428.

**Pull request merged**